extensions/cursor-directory/src/index.tsx

```diff
@@ -66,111 +74,108 @@ export default function Command() {
-                                  {cursorRule.tags.slice(0, 3).map((tag) => (
-                                    <List.Item.Detail.Metadata.TagList.Item key={tag} text={tag} />
-                                  ))}
-                                </List.Item.Detail.Metadata.TagList>
-                              </>
-                            )}
-                            {cursorRule?.libs && cursorRule.libs.length > 0 && (
-                              <List.Item.Detail.Metadata.TagList title="Libraries">
-                                {cursorRule.libs.slice(0, 3).map((lib) => (
-                                  <List.Item.Detail.Metadata.TagList.Item key={lib} text={lib} />
-                                ))}
-                              </List.Item.Detail.Metadata.TagList>
-                            )}
-                          </List.Item.Detail.Metadata>
-                        }
-                      />
-                    ),
+                    detail: <CursorRulePreview cursorRule={cursorRule} section={section} popularOnly={popularOnly} />,
                   }
                 : {
                     accessories: [
-                      { text: cursorRule?.tags.slice(0, 3).join(", ") },
-                      ...(popularOnly && cursorRule?.count
+                      { text: cursorRule.tags.slice(0, 3).join(", ") },
+                      ...(popularOnly && cursorRule.count !== null
                         ? [{ icon: Icon.Person, text: cursorRule.count.toString() }]
                         : []),
+                      { icon: Icon.StarDisabled },
                     ],
                   };
 
               return (
                 <List.Item
-                  key={cursorRule?.slug || index}
-                  title={cursorRule?.title || ""}
+                  key={cursorRule.slug}
+                  title={cursorRule.title}
                   {...props}
                   actions={
                     <ActionPanel>
                       <ActionPanel.Section title="Actions">
-                        <Action.Push
-                          title="View Cursor Rule"
-                          icon={Icon.Text}
-                          target={<CursorRuleDetail cursorRule={cursorRule!} popularOnly={popularOnly} />}
-                        />
+                        <ViewRuleAction cursorRule={cursorRule} popularOnly={popularOnly} />
+                        <CopyRuleAction cursorRule={cursorRule} />
+                        <ToggleViewAction showingDetail={showingDetail} setShowingDetail={setShowingDetail} />
                         <Action
-                          title="Copy Cursor Rule"
-                          icon={Icon.Clipboard}
-                          shortcut={{ modifiers: ["cmd"], key: "c" }}
+                          title="Star Cursor Rule"
+                          icon={Icon.Star}
+                          shortcut={{ modifiers: ["cmd"], key: "s" }}
```

Comment:
In Raycast we use the same hotkey for both unpin/pin - you can use `Keyboard.Shortcut.Common.Pin`